wandb/integration/openai/fine_tune.py

```diff
@@ -0,0 +1,352 @@
+                wandb.termwarn("Fine-tuning finished, logging metrics to W&B")
+                return fine_tune
+            if fine_tune.status == "failed":
+                wandb.termwarn(
+                    "Fine-tune {fine_tune_id} has has failed and will not be logged"
+                )
+                return fine_tune
+            if fine_tune.status == "cancelled":
+                wandb.termwarn(
+                    "Fine-tune {fine_tune_id} has was cancelled and will not be logged"
+                )
+                return fine_tune
+            time.sleep(60)
+            fine_tune = cls.openai_client.fine_tuning.jobs.retrieve(
+                fine_tuning_job_id=fine_tune.id
+            )
+
+    @classmethod
+    def _log_fine_tune(
+        cls,
+        fine_tune,
+        project,
+        entity,
+        force,
+        show_individual_warnings=None,
+        **kwargs_wandb_init,
+    ):
+        fine_tune_id = fine_tune.id
+        status = fine_tune.status
+
+        # check run completed successfully
+        if status != "succeeded":
+            if show_individual_warnings:
+                print(
+                    f'Fine-tune {fine_tune_id} has the status "{status}" and will not be logged'
+                )
+            return
+
+        # check results are present
+        try:
+            results_id = fine_tune.result_files[0]
+            results = cls.openai_client.files.retrieve_content(file_id=results_id)
+        except openai.NotFoundError:
+            if show_individual_warnings:
+                print(f"Fine-tune {fine_tune_id} has no results and will not be logged")
+            return
+
+        # check run has not been logged already
+        run_path = f"{project}/{fine_tune_id}"
+        if entity is not None:
+            run_path = f"{entity}/{run_path}"
+        wandb_run = cls._get_wandb_run(run_path)
+        if wandb_run:
+            wandb_status = wandb_run.summary.get("status")
+            if show_individual_warnings:
+                if wandb_status == "succeeded":
+                    print(
+                        f"Fine-tune {fine_tune_id} has already been logged successfully at {wandb_run.url}"
+                    )
+                    if not force:
+                        print(
+                            'Use "--force" in the CLI or "force=True" in python if you want to overwrite previous run'
+                        )
+                else:
+                    print(
+                        f"A run for fine-tune {fine_tune_id} was previously created but didn't end successfully"
+                    )
+                if wandb_status != "succeeded" or force:
+                    print(
+                        f"A new wandb run will be created for fine-tune {fine_tune_id} and previous run will be overwritten"
+                    )
+            if wandb_status == "succeeded" and not force:
+                return
+
+        # start a wandb run
+        wandb.init(
```

Comment:
wait if you already had a wandb_run, could you leave a comment explaining why you start another one?

Reply:
also could you assign the result into a value and use the run object. i.e.:

run = wandb.init(...